engine/api/project/dao_variable.go

```diff
@@ -0,0 +1,212 @@
+		WHERE project_id = $1
+		ORDER BY var_name
+			  `).Args(projID)
+	return loadAllVariables(db, query)
+}
+
+// LoadAllVariablesWithDecrytion Get all variable for the given project, it also decrypt all the secure content
+func LoadAllVariablesWithDecrytion(db gorp.SqlExecutor, projID int64) ([]sdk.Variable, error) {
+	query := gorpmapping.NewQuery(`
+		SELECT *
+		FROM project_variable
+		WHERE project_id = $1
+		ORDER BY var_name
+			  `).Args(projID)
+	return loadAllVariables(db, query, gorpmapping.GetOptions.WithDecryption)
+}
+
+func loadVariable(db gorp.SqlExecutor, q gorpmapping.Query, opts ...gorpmapping.GetOptionFunc) (*sdk.Variable, error) {
+	var v dbProjectVariable
+	found, err := gorpmapping.Get(context.Background(), db, q, &v, opts...)
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, sdk.WithStack(sdk.ErrNotFound)
+	}
+	isValid, err := gorpmapping.CheckSignature(v, v.Signature)
+	if err != nil {
+		return nil, err
+	}
+	if !isValid {
+		log.Error(context.Background(), "project.loadVariable> project variable %d data corrupted", v.ID)
+		return nil, sdk.WithStack(sdk.ErrNotFound)
+	}
+
+	res := v.Variable()
+	return &res, err
+}
+
+// LoadVariable retrieve a specific variable
+func LoadVariable(db gorp.SqlExecutor, projID int64, varName string) (*sdk.Variable, error) {
+	query := gorpmapping.NewQuery(`SELECT * FROM project_variable
+			WHERE project_id = $1 AND var_name=$2`).Args(projID, varName)
+	return loadVariable(db, query)
+}
+
+// LoadVariableWithDecryption retrieve a specific variable with decrypted content
+func LoadVariableWithDecryption(db gorp.SqlExecutor, projID int64, varID int64, varName string) (*sdk.Variable, error) {
+	query := gorpmapping.NewQuery(`SELECT * FROM project_variable
+			WHERE project_id = $1 AND id = $2 AND var_name=$3`).Args(projID, varID, varName)
+	return loadVariable(db, query, gorpmapping.GetOptions.WithDecryption)
+}
+
+// DeleteAllVariables Delete all variables from the given project.
+func DeleteAllVariables(db gorp.SqlExecutor, projectID int64) error {
+	query := `DELETE FROM project_variable
+	          WHERE project_variable.project_id = $1`
+	if _, err := db.Exec(query, projectID); err != nil {
+		return sdk.WithStack(err)
+	}
+	return nil
+}
+
+// InsertVariable Insert a new variable in the given project
+func InsertVariable(db gorp.SqlExecutor, projID int64, v *sdk.Variable, u sdk.Identifiable) error {
+	//Check variable name
+	rx := sdk.NamePatternRegex
+	if !rx.MatchString(v.Name) {
+		return sdk.NewError(sdk.ErrInvalidName, fmt.Errorf("Invalid variable name. It should match %s", sdk.NamePattern))
+	}
+
+	if sdk.NeedPlaceholder(v.Type) && v.Value == sdk.PasswordPlaceholder {
+		return fmt.Errorf("You try to insert a placeholder for new variable %s", v.Name)
+	}
+
+	dbVar := newDBProjectVariable(*v, projID)
+
+	err := gorpmapping.InsertAndSign(context.Background(), db, &dbVar)
+	if err != nil && strings.Contains(err.Error(), "project_variable_pkey") {
```

Comment:
No err type for this purpose ?

Reply:
you're right it was a weird existing err management